### PR TITLE
Docs: Add Azure, Filesystem, Imgur and S3

### PR DESCRIPTION
**Pull request by DerMolly**
### Component/Part
docs

### Description
This PR adds documentation for the media backends

- Azure
- Filesystem
- Imgur
- S3

### Steps

- [x] Added / updated documentation
- [x] I read the [contribution documentation](https://github.com/hedgedoc/hedgedoc/blob/develop/CONTRIBUTING.md) and
  made sure that:
  - My commits are signed-off to accept the DCO.
  - This PR targets the correct branch: `master` for 1.x & docs, `develop` for 2.x

### CODE REVIEW

File: docs/content/media/azure.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+## Azure
+
+You can use [Microsoft Azure Blob Storage](https://azure.microsoft.com/services/storage/blobs/) to handle your image uploads in HedgeDoc.
+
+All you need to do is create a container with public access set to 'blob' and a [connection string](https://docs.microsoft.com/azure/storage/common/storage-account-keys-manage).
+
+It's possible to create the container with the [azure cli](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli), if you have a connection string, with the following command:
+
+```
+az storage container create --name NAME --public-access blob --connection-string "CONNECTION_STRING"
```
Comment (InnayTool) [inline docs/content/media/azure.md:10]: Maybe the templates should be annotated in another way. Maybe like this:
```suggestion
az storage container create --name NAME --public-access blob --connection-string "<CONNECTION_STRING>"
```
This is especially relevant for the environment variables.

**Review by InnayTool — APPROVED** [review]

**Review by InnayTool — COMMENTED** [review]
See the annotation above.

File: docs/content/media/azure.md (hunk 1 above)
Comment (DerMolly) [reply docs/content/media/azure.md:10]: Should this be changed for all placeholder or only that one?

Comment (InnayTool) [reply docs/content/media/azure.md:10]: For all, maybe there is a better way but I think it's important to see clearly what is a placeholder.

Comment (DerMolly) [reply docs/content/media/azure.md]: fixed

**Review by InnayTool — DISMISSED** [review]

File: docs/content/media/s3.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+## S3
+
+You can use [Amazon S3](https://aws.amazon.com/s3/) or any other S3 compatible storage to handle your image uploads in HedgeDoc.
```
Comment (davidmehren) [inline docs/content/media/s3.md:3]:
```suggestion
You can use [Amazon S3](https://aws.amazon.com/s3/) or any other S3-compatible storage (like [MinIO](https://min.io) or [Ceph Object Gateway](https://docs.ceph.com/en/latest/radosgw/)) to handle your image uploads in HedgeDoc.
```

We should probably mention some alternatives here and provide an example config?

**Review by davidmehren — CHANGES_REQUESTED** [review]

File: docs/content/media/azure.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
+## Azure
```
[the remark this answers is not in the document]
Comment (DerMolly) [reply docs/content/media/azure.md:1]: done

File: docs/content/media/azure.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+You can use [Microsoft Azure Blob Storage](https://azure.microsoft.com/services/storage/blobs/) to handle your image uploads in HedgeDoc.
+
+All you need to do is create a container with public access set to 'blob' and a [connection string](https://docs.microsoft.com/azure/storage/common/storage-account-keys-manage).
```
[the remark this answers is not in the document]
Comment (DerMolly) [reply docs/content/media/azure.md:5]: done

File: docs/content/media/azure.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+It's possible to create the container with the [azure cli](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli), if you have a connection string, with the following command:
```
[the remark this answers is not in the document]
Comment (DerMolly) [reply docs/content/media/azure.md:7]: done

File: docs/content/media/filesystem.md [hunk 6]
```diff
@@ -0,0 +1,12 @@
+## Filesystem
```
[the remark this answers is not in the document]
Comment (DerMolly) [reply docs/content/media/filesystem.md:1]: done

File: docs/content/media/imgur.md [hunk 7]
```diff
@@ -0,0 +1,17 @@
+## Imgur
```
[the remark this answers is not in the document]
Comment (DerMolly) [reply docs/content/media/imgur.md:1]: done

File: docs/content/media/imgur.md [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+
+You can use [Imgur](https://imgur.com) to handle your image uploads in HedgeDoc.
```
[the remark this answers is not in the document]
Comment (DerMolly) [reply docs/content/media/imgur.md:3]: done

File: docs/content/media/imgur.md [hunk 9]
```diff
@@ -0,0 +1,17 @@
+## Imgur
+
+You can use [Imgur](https://imgur.com) to handle your image uploads in HedgeDoc.  
+All you need for that is to register an application with Imgur and get a client id:
+1. Create an account on imgur.com and log in.
+2. Go to https://api.imgur.com/oauth2/addclient
```
[the remark this answers is not in the document]
Comment (DerMolly) [reply docs/content/media/imgur.md:6]: done via `<>`

File: docs/content/media/s3.md [hunk 10]
```diff
@@ -0,0 +1,17 @@
+## S3
```
[the remark this answers is not in the document]
Comment (DerMolly) [reply docs/content/media/s3.md:1]: done

File: docs/content/media/s3.md [hunk 11]
```diff
@@ -0,0 +1,17 @@
...
+
+You can use [Amazon S3](https://aws.amazon.com/s3/) or any other S3 compatible storage to handle your image uploads in HedgeDoc.
+
+The BUCKET should be configured to be writeable.
```
[the remark this answers is not in the document]
Comment (DerMolly) [reply docs/content/media/s3.md:5]: done

File: docs/content/media/s3.md [hunk 12]
```diff
@@ -0,0 +1,17 @@
...
+<small>(with the appropriate substitution for `<ACCESS_KEY>`, `<SECRET_KEY>`, `<BUCKET>`, and `<ENDPOINT>` of course)</small>
+```
+HD_MEDIA_BACKEND="s3"
+HD_MEDIA_BACKEND_S3_ACCESS_KEY="<ACCESS_KEY>"
+HD_MEDIA_BACKEND_S3_SECRET_KEY="<SECRET_KEY>"
+HD_MEDIA_BACKEND_S3_BUCKET="<BUCKET>"
+HD_MEDIA_BACKEND_S3_ENDPOINT="<ENDPOINT>"
+```
+
+If you use Amazon S3, `<ENDPOINT>` should contain your [Amazon Region](https://docs.aws.amazon.com/general/latest/gr/s3.html).
```
[the remark this answers is not in the document]
Comment (DerMolly) [reply docs/content/media/s3.md:17]: added

File: docs/content/media/s3.md (hunk 2 above)
Comment (DerMolly) [reply docs/content/media/s3.md:3]: Not sure what example config we should provide here? I added the suggested change.

File: docs/content/media/azure.md [hunk 13]
```diff
@@ -0,0 +1,21 @@
+# Azure Blob Storage
+
+You can use [Microsoft Azure Blob Storage](https://azure.microsoft.com/services/storage/blobs/) to handle your image uploads in HedgeDoc.
+
+All you need to do is get the [connection string](https://docs.microsoft.com/azure/storage/common/storage-account-keys-manage) for your storage account and create a storage container with public access set to 'blob'.
```
Comment (davidmehren) [inline docs/content/media/azure.md:5]:
```suggestion
All you need to do is to get the [connection string](https://docs.microsoft.com/azure/storage/common/storage-account-keys-manage) for your storage account and create a storage container with public access set to 'blob'.
```

File: docs/content/media/azure.md [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+
+It's possible to create the container with the [Azure CLI](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli), using your connection string, with the following command:
+
+```
+az storage container create --name <NAME> --public-access blob --connection-string "<CONNECTION_STRING>"
+```
+
+You can of course also create the container in the azure portal if you prefer.
```
Comment (davidmehren) [inline docs/content/media/azure.md:13]:
```suggestion
You can of course also create the container in the Azure portal if you prefer.
```

File: docs/content/media/s3.md [hunk 15]
```diff
@@ -0,0 +1,19 @@
...
+
+For example: If you're Amazon Region was `us-east-2` your endpoint `<ENDPOINT>` should be `s3.us-east-2.amazonaws.com`.
```
Comment (davidmehren) [inline docs/content/media/s3.md:19]:
```suggestion
For example: If your Amazon Region is `us-east-2`, your endpoint `<ENDPOINT>` should be `s3.us-east-2.amazonaws.com`.
```

**Review by davidmehren — CHANGES_REQUESTED** [review]
Found a typo in the new example and two minor things I missed on the first review.

File: docs/content/media/azure.md (hunk 13 above)
Comment (DerMolly) [reply docs/content/media/azure.md:5]: done

File: docs/content/media/azure.md (hunk 14 above)
Comment (DerMolly) [reply docs/content/media/azure.md:13]: done

File: docs/content/media/s3.md (hunk 15 above)
Comment (DerMolly) [reply docs/content/media/s3.md:19]: done

**Review by davidmehren — APPROVED** [review]

**Review by InnayTool — APPROVED** [review]

**Pull request merged**